docs/concepts.rst

```diff
@@ -0,0 +1,153 @@
+.. _ksql_concepts:
+
+Concepts
+========
+
+==========
+Components
+==========
+
+The main components of KSQL are the KSQL CLI and the KSQL server.
+
+KSQL CLI
+--------
+
+The KSQL CLI allows you to interactively write KSQL queries. Its
+interface should be familiar to users of MySQL, Postgres, Oracle, Hive,
+Presto, etc.
+
+The KSQL CLI acts as a client to the KSQL server.
+
+KSQL Server
+-----------
+
+The KSQL server runs the engine that executes KSQL queries, which
+includes the data processing as well as reading data from and writing
+data to the target Kafka cluster.
+
+===========
+Terminology
+===========
+
+When using KSQL, the following terminology is used.
+
+Stream
+------
+
+A stream is an unbounded sequence of structured data (“facts”). For
+example, we could have a stream of financial transactions such as “Alice
+sent $100 to Bob, then Charlie sent $50 to Bob”. Facts in a stream are
+immutable, which means new facts can be inserted to a stream, but
+existing facts can never be updated or deleted. Streams can be created
+from a Kafka topic or derived from existing streams and tables. In both
```

Comment:
Streams cannot be created from tables.

Reply:
@hjafarpour so should this portion be deleted `or derived from existing streams and tables`?

Reply:
It can be `derived from existing streams.`